### ISSUE TITLE
Memoize functions returned by useDebouncedCallback (i.e. by useCallback)

### ISSUE THREAD

**Issue by jack-sf** [xnimorz/use-debounce#18]
Let's say I have the same code as the one stated in current README of use-debounce:

```js
  const [scrollHandler] = useDebouncedCallback(
    () => { setPosition(window.pageYOffset); },
    800,
    []
  );

  useEffect(() => {
    const unsubscribe = subscribe(window, 'scroll', scrollHandler);
    return () => {
      unsubscribe();
    };
  }, []);
```

But if I have eslint react [exhaustive-deps](https://reactjs.org/docs/hooks-reference.html#conditionally-firing-an-effect) rule rule enabled, it gives me the following warning:

![image](https://user-images.githubusercontent.com/44373826/55718613-073f6c00-59f4-11e9-8802-07639e6e9833.png)

To fix the eslint warning, I have to add the debounced function as a dependency of the useEffect:

```js
  const [scrollHandler] = useDebouncedCallback(
    () => { setPosition(window.pageYOffset); },
    800,
    []
  );

  useEffect(() => {
    const unsubscribe = subscribe(window, 'scroll', scrollHandler);
    return () => {
      unsubscribe();
    };
  }, [scrollHandler]);
```

But such code will return a new reference to `scrollHandler` on every render, thus forcing the effect to be subscribed and unsubscribed on each render.

So to fix it I actually have to do sth like:

```js
  const [scrollHandler] = useDebouncedCallback(
    () => { setPosition(window.pageYOffset); },
    800,
    []
  );
  const memoizedScrollHandler = useCallback(scrollHandler, []);

  useEffect(() => {
    const unsubscribe = subscribe(window, 'scroll', memoizedScrollHandler);
    return () => {
      unsubscribe();
    };
  }, [memoizedScrollHandler]);
```

Then everything works well.

But we could avoid requiring me to write this boilerplate of 

```js
const memoizedScrollHandler = useCallback(scrollHandler, [])
```

Instead, useDebouncedCallback could return functions that are already wrapped with `useCallback`, and they return a new function value only when some deps or delay is changed.

For example, it seems like `cancelDebouncedCallback` is already wrapped with `useCallback` ( https://github.com/xnimorz/use-debounce/blob/master/src/callback.js#L12 ).

WDYT?

**Comment by xnimorz** [comment]
Hello, @jack-sf 
Thank you for this issue. 

I'll memoize debouncedCallback so that you won't use the boilerplate:
```
const memoizedScrollHandler = useCallback(scrollHandler, [])
```

I'll write here when I cut the new release (I think I'll do it within 1-2 days)

**Comment by xnimorz** [comment]
Hi @jack-sf .

I've just published the new (1.1.2) version https://github.com/xnimorz/use-debounce/commit/5e65f4a387999a6d15a400b850201a67a46ca3a6 which includes `debouncedCallback` memoization.

Thank you again for the idea!

**Comment by jack-sf** [comment]
Super! Now it works perfectly. Thanks for quick reaction ;)
